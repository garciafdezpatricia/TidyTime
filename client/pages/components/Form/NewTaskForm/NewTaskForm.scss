@use "../../../../styles/variables" as *;

.create-task-form {
    display: flex;
    width: 80vw;
    position: absolute;
    bottom: 2rem;
    left: 2rem;
    input {
        padding-top: .2rem;
        padding-bottom: .2rem;
        padding-left: 1rem;
        width: 90%;
        outline: none;
        border: 2px solid #7c7c7c;
        border-radius: 4px;
        &:active, &:focus {
            outline: none;
            border: 2px solid $primary-green;
        }
    }
}