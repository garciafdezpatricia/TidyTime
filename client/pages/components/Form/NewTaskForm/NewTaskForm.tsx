import { useRef, useState } from "react";
import { FaPlusCircle } from "react-icons/fa";
import { IoIosAddCircle } from "react-icons/io";

export interface Props {
    onSubmit: (arg?:any) => void | any;
}

export default function NewTaskForm(props : Props) {

    // use a reference instead of state: we don't want to rerender everytime we type in the input
    const newTask = useRef(null);
    const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
        // Verificar si la tecla presionada es Enter (código de tecla 13)
        if (e.key === 'Enter') {
            e.preventDefault(); // Evitar el comportamiento predeterminado de enviar el formulario
            createNewTask();
        }
    };
    
    const createNewTask = () => {
        // @ts-ignore
        if (newTask.current.value !== '') {
           // @ts-ignore
        props.onSubmit(newTask.current.value); // Llamar a la función onSubmit pasada como prop
        // @ts-ignore
        newTask.current.value = '';
        }
    }

    return (
        <form className="create-task-form" >
            {// TODO: add onclick handler to the icon (create task too)
            }
            <IoIosAddCircle 
                size={"1.5rem"} 
                color={"#3E5B41"}
                cursor={"pointer"}
                onClick={createNewTask}/>
            <input 
                placeholder="New task..."
                ref={newTask} 
                type="text"
                onKeyDown={handleKeyPress}
            >
            </input>
        </form>
    )
}