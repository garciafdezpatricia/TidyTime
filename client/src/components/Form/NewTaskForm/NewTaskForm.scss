@use "../../../../styles/variables" as *;

.create-task-form {
    display: flex;
    gap: .8rem;
    align-items: center;
    input {
        padding-top: .2rem;
        padding-bottom: .2rem;
        padding-left: .2rem;
        width: 90%;
        outline: none;
        border: 2px solid #7c7c7c;
        border-radius: 4px;
        &:active, &:focus {
            outline: none;
            border: 2px solid $primary-green;
        }

        @media screen and (min-width: 320px) and (max-width: 768px) {
            font-size: .7rem;
        }

        @media screen and (min-width: 769px) and (max-width: 1200px) {
            font-size: .7rem;
            width: 80%;
        }
    }

    @media screen and (min-width: 320px) and (max-width: 768px) {
        flex-direction: row-reverse;
        gap: .3rem;
    }
}

body.dark-mode .create-task-form input {
    background-color: $background-dark-1;
    color: $white;
}