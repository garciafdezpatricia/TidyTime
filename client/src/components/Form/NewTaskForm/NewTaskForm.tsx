import { useRef, useState } from "react";
import { FaPlusCircle } from "react-icons/fa";
import { IoIosAddCircle } from "react-icons/io";
import { useTaskContext } from "../../Context/TaskContext";

export default function NewTaskForm() {

    // use a reference instead of state: we don't want to rerender everytime we type in the input
    const newTask = useRef(null);
    const {listNames, tasks, selectedListIndex, setListNames, setTasks} = useTaskContext();

    const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
        // Verificar si la tecla presionada es Enter (código de tecla 13)
        if (e.key === 'Enter') {
            e.preventDefault(); // Evitar el comportamiento predeterminado de enviar el formulario
            createNewTask();
        }
    };
    
    const createNewTask = () => {
        // @ts-ignore
        if (newTask.current.value !== '') {
            // @ts-ignore
            addNewTask(newTask.current.value); // Llamar a la función onSubmit pasada como prop
            // @ts-ignore
            newTask.current.value = '';
        }
    }

    const addNewTask = (e: any) => {
		setTasks((prevTodo) => {
			return prevTodo.map((list, index) => {
				if (index === selectedListIndex) {
                    return [{ title: e, done: false, listIndex: selectedListIndex }, ...list];
                } 
                return list;
            });
        });
	}

    return (
        <form className="create-task-form" >
            <IoIosAddCircle 
                size={"1.5rem"} 
                color={"#3E5B41"}
                cursor={"pointer"}
                onClick={createNewTask}/>
            <input 
                placeholder="New task..."
                ref={newTask} 
                type="text"
                onKeyDown={handleKeyPress}
            >
            </input>
        </form>
    )
}