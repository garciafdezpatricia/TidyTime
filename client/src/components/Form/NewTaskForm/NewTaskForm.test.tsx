import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import NewTaskForm from '@/src/components/Form/NewTaskForm/NewTaskForm';

describe('NewTaskForm', () => {
    it('renders without crashing', () => {
      const { container } = render(<NewTaskForm/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<NewTaskForm/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });