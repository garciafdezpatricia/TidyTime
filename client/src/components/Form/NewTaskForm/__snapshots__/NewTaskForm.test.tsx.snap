// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskForm matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="create-task-form"
  >
    <svg
      color="#000000"
      cursor="pointer"
      data-testid="create-task-button"
      fill="currentColor"
      height="1.5rem"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(0, 0, 0);"
      viewBox="0 0 512 512"
      width="1.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm90.5 224H272v74.5c0 8.8-7.2 16-16 16-4.4 0-8.4-1.8-11.3-4.7-2.9-2.9-4.7-6.9-4.7-11.3V272h-74.5c-4.4 0-8.4-1.8-11.3-4.7-2.9-2.9-4.7-6.9-4.7-11.3 0-8.8 7.2-16 16-16H240v-74.5c0-8.8 7.2-16 16-16s16 7.2 16 16V240h74.5c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
      />
    </svg>
    <input
      placeholder="list.newTask"
      type="text"
    />
  </form>
</DocumentFragment>
`;
